### ISSUE TITLE
go-yaml v1.13.7 and go-yaml v1.13.8 parse error

### ISSUE THREAD

**Issue by yxxhero** [goccy/go-yaml#528]
**Describe the bug**

**To Reproduce**

Please provide a minimum yaml content that can be reproduced.
We are more than happy to use [Go Playground](https://go.dev/play/p/cG_sFccgqTk )

```
package main

import (
	"fmt"

	"github.com/goccy/go-yaml"
)

func main() {
	yamlC := `helmfiles:
  - path: subhelmfile.yaml.gotmpl
    selectorsInherited: true
    values:
    - specificValues:
        
        myContainers:
        - command:
          - /bin/sh
          - -c
          - env
          image: registry.k8s.io/busybox
          name: test-container
        
`
	target := make(map[string]any)

	err := yaml.Unmarshal([]byte(yamlC), &target)
	if err != nil {
		fmt.Printf("%v", err)
	} else {
		fmt.Print("ok")
	}
}

```

**Expected behavior**

**Screenshots**

**Version Variables**
 - Go version: 1.23
 - go-yaml's Version: 1.13.8

**Additional context**

**Comment by goccy** [comment]
@yxxhero Thank you for your reports, but I couldn't reproduce from the provided example code.
Please provide a full yaml content that can be reproduced. Also, I'm more than happy to use [Go Playground](https://go.dev/play)

**Comment by yxxhero** [comment]
@goccy thanks so much for your feedback. I use https://go.dev/play/p/cG_sFccgqTk to reproduce this issue. please review. thanks so much.

**Comment by yxxhero** [comment]
there are some extra space char in the end. but it's right yaml format.

**Comment by yxxhero** [comment]
@goccy thanks so much.

**Comment by goccy** [comment]
Thank you for your additional information ! I've found this problem and fixed with #529 !
